linera-core/src/client/mod.rs

```diff
@@ -2091,9 +1968,11 @@ where
         let incoming_bundles = self.pending_message_bundles().await?;
         let identity = self.identity().await?;
         let info = self.chain_info_with_manager_values().await?;
-        let round = match Self::round_for_new_proposal(&info, &identity, None)? {
-            Either::Left(round) => round,
-            Either::Right(timeout) => return Ok(ExecuteBlockOutcome::WaitForTimeout(timeout)),
+        let Some(round) = Self::round_for_new_proposal(&info.manager, &identity, None) else {
+            let timeout = info.round_timeout().ok_or_else(|| {
+                ChainClientError::BlockProposalError("Cannot propose in the current round.")
```

Comment:
Is there a specific reason why we can't propose here? If there is, can we make that an error variant (or even put the reason in the msg)?